pkg/cmd/codespace/view.go

```diff
@@ -0,0 +1,98 @@
+package codespace
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strconv"
+
+	"github.com/cli/cli/v2/internal/codespaces/api"
+	"github.com/cli/cli/v2/pkg/cmdutil"
+	"github.com/cli/cli/v2/utils"
+	"github.com/spf13/cobra"
+)
+
+type viewOptions struct {
+	selector *CodespaceSelector
+}
+
+func newViewCmd(app *App) *cobra.Command {
+	opts := &viewOptions{}
+	var exporter cmdutil.Exporter
+
+	viewCmd := &cobra.Command{
+		Use:   "view",
+		Short: "View details about a codespace",
+		Args:  noArgsConstraint,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return app.ViewCodespace(cmd.Context(), opts, exporter)
+		},
+	}
+	opts.selector = AddCodespaceSelector(viewCmd, app.apiClient)
+	cmdutil.AddJSONFlags(viewCmd, &exporter, api.ViewCodespaceFields)
+
+	return viewCmd
+}
+
+func (a *App) ViewCodespace(ctx context.Context, opts *viewOptions, exporter cmdutil.Exporter) error {
+	// If we are in a codespace, show the details for it
+	if codespaceName := os.Getenv("CODESPACE_NAME"); os.Getenv("CODESPACES") == "true" && codespaceName != "" {
+		opts.selector.codespaceName = codespaceName
```

Comment:
Should we only do this if `opts.selector.codespaceName` is empty (i.e. the user didn't specify it)

Reply:
Yep that's a good idea. Should we do this by default for all commands that use the selector or save it for just this?

Reply:
I didn't realize other command had this pattern - if so, they should also be fixed to respect the -c param. It's fine to do that separately though

Reply:
It's not used anywhere else at the moment! I meant adding the in-codespace check [here](https://github.com/cli/cli/blob/570a7202c3b331a9fdc39508d3a754c6847b3513/pkg/cmd/codespace/codespace_selector.go#L43) for all uses of the selector rather than just for the `view` command.

Reply:
Oh I see!

I think there are some commands that this behavior would be iseful, but definitely don't want it as the default:

- `cs ssh`, `cs code`, etc: you're already in the codespace so probably not super useful
- `cs ports`: actually could be really useful
- `cs rebuild`: could be useful in ssh mode, but not sure if it would cause the ssh session to end?
- `cs stop`: also could be useful in ssh mode as an "exit and shutdown" helper
- `cs rename`: would work, but probably not a super common scenario
- `cs delete`: if you run that command in a codespace it suddenly just deletes the codespace you're currently working in 😅

Reply:
In that case I will hold off on updating the other commands!